duneprototypes/Protodune/hd/RawDecoding/pdhd_daphne2_decoder.fcl

```diff
@@ -0,0 +1,43 @@
+#include "DAPHNEReaderPDHD.fcl"
+#include "HDF5RawInput2.fcl"
+
+services:
+{
+  TimeTracker:  {}
+  HDF5RawFile2Service:  {}
+  TFileService: {
+    fileName: "pdhd_daphne_decodermodule.root"
+  } 
+  DAPHNEChannelMapService: {
+    #FileName: "DAPHNE_test_ChannelMap_v0.txt"
+    FileName: "DAPHNE_test5_ChannelMap_v1.txt"
+    IgnoreLinks: true
+  }
+}
+
+physics:
+{
+  producers:
+  {
+    pdhddaphne: @local::DAPHNEReaderPDHD
+  }
+
+  produce: [ pdhddaphne ] 
+  output: [ out1 ]
+  trigger_paths : [ produce ]
+  end_paths: [ output ]
+}
+
+outputs:
+{
+  out1:
+  {
+    compressionLevel: 1
+    module_type: RootOutput
+    fileName: "%ifb_daphne.root"
+  }
+}
+
+source: @local::hdf5rawinput2
```

Comment:
Oh, and you'll need to use hdf5rawinput3 here for the source.

 I think I see an overall design choice that's different here.  The DAQ people changed the trigger record header size, which has ripple effects through HDF5RawInputFile which is the class we use to retrieve the data.  I wanted to keep backward compatibility, but also have the ability to read new data too.  Since I was not eager to rewrite the DAQ code in hdf5libs to do what is done here -- to automatically detect the version and do the right thing, instead I made a separate copy which includes the new headers.

This new copy was the right thing when we moved from artROOT to HDF5, and also from the first HDF5 format to the second, given how different they were; it was all but impossible to try to write a source and decoder that could understand both artROOT and HDF5, or even the two kinds of HDF5 the DAQ people wrote.  This version is a much smaller update, but it still is breaking.  I took the same approach, to separate out new source, service and decoder, while retaining the old ones for backward compatibility, and one has to use the right one when reading old data or new data, it does not automatically choose.

The DAQ group changed the trigger record header in the same overall release as the DAPHNE upgrade, and so I don't think you'll ever see a mixture of DAPHNE2 and DAPHNE3's with the same HDF5 format.  So for the TPC decoders, they only decode one format at a time, and if we need to deal with another format, we just make another decoder.